### PR TITLE
Fix various errors in try client with Python 3

### PR DESCRIPTION
**Pull request by gbenson**
## Contributor Checklist:
* [X] I have updated the unit tests
* [X] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by gbenson** [comment]
FWIW I checked 'I have updated the appropriate documentation', but I don't know there is any other than the news fragment...

**Comment by p12tic** [comment]
Bug fixes don't need documentation updates, so everything is perfect.

File: master/buildbot/newsfragments/try-client-python3-fixes.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+:issue:`4765`: Fix various errors in try client with Python 3
```
Comment (p12tic) [inline master/buildbot/newsfragments/try-client-python3-fixes.bugfix]: I think this should've been `4746`, because 4765 points to a PR.

**Comment by gbenson** [comment]
> Thanks a lot for the contribution!

Thanks for making it easy to contribute :D

**Pull request merged**